subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/api/ArtifactDeclarationIntegrationTest.groovy

```diff
@@ -317,6 +317,97 @@ task checkArtifacts {
         result.assertTasksExecuted(":a:classes", ":b:checkArtifacts")
     }
 
+    def "can define artifact using flatMap of Task provider"() {
+        settingsFile << "include 'a', 'b'"
+        buildFile << """
+            class Producer extends DefaultTask {
+                @OutputFile
+                RegularFileProperty outputFile = project.objects.fileProperty()
+                
+                @TaskAction
+                void makeFile() {
+                    outputFile.get().asFile.text = "foobar"
+                }
+            }
+            
+            project(':a') {
+                def producer = tasks.register("producer", Producer) {
+                    outputFile = layout.buildDirectory.file("foobar.txt")
+                }
+                
+                artifacts {
+                    compile producer.flatMap { it.outputFile }
```

Comment:
so if I have a `Provider<? extends AbstractArchiveTask>` what would this look like?  With a realized task, I can just say `compile jarTask`.  Is there a reasonable way to declare an artifact backed by a provider of an `AbstractArchiveTask`?